qubes-gui-client/src/buffer.rs

```diff
@@ -215,7 +237,7 @@ impl<T: VchanMock> RawMessageStream<T> {
     /// more data needs to arrive, returns `Ok(None)`.  If an error occurs,
     /// `Err` is returned, and the stream is placed in an error state.  If the
     /// stream is in an error state, all further functions will fail.
-    pub fn read_message<'a, 'b: 'a>(&'b mut self) -> io::Result<Option<(Header, &'a [u8])>> {
+    pub fn read_message<'a, 'b: 'a>(&'b mut self) -> io::Result<Option<ClaimableBuffer<'a, T>>> {
```

Comment:
I don't think you don't need to use two lifetimes here, as calling a method automatically inserts a reborrow that can shrink the lifetime of the `&mut` passed, so there's no need to call out such a possibility explicitly.

In fact, writing this as `pub fn read_message<'a>(&'a mut self) -> io::Result<Option<ClaimableBuffer<'a, T>>>`.